cap-containers/map.h

```diff
@@ -44,6 +44,14 @@ typedef struct cap_map {
 	struct cap_map *_forward[CAP_MAP_MAX_SKIPLIST_SIZE];
 } cap_map;
 
+// Since on the Iterator for cap_map we are traversing through the singly linked list, we do not need to store an array for all elements, We only need a single _current pointer which will point to the current element which we are seeing(initially it will be map->_forward[0]). If the cap_map is empty, the item with index 0 will be NULL, in that case, the _key and _value should also be NULL, we expect the user to to check against NULL.
+typedef struct {
+	CAP_GENERIC_TYPE_PTR _key;
+	void *_value;
+	size_t _current_index;
+	cap_map *current_element;
```

Comment:
since the user is not supposed to access the `current_element`, which is a private member meaning, it's internally stored to help with the implementation. Please write a leading `_` for private members, like `_current_element`.

Also, we assume that a user will access the `key` and `value` of a node. So make them public meaning *without* the leading `_` like `void *value; CAP_GENERIC_TYPE_PTR key`